node/state/core.go

```diff
@@ -0,0 +1,28 @@
+package state
+
+import (
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/tendermint/spm/cosmoscmd"
+
+	"github.com/celestiaorg/celestia-app/app"
+	apptypes "github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/celestiaorg/celestia-node/node/services"
+	"github.com/celestiaorg/celestia-node/params"
+	"github.com/celestiaorg/celestia-node/service/state"
+)
+
+func CoreAccessor(
+	keyCfg services.KeyConfig,
+	storePath string,
+	coreEndpoint string,
+) (state.Accessor, error) {
+	encCfg := cosmoscmd.MakeEncodingConfig(app.ModuleBasics)
+
+	ring, err := keyring.New(app.Name, keyring.BackendFile, storePath, nil) // TODO @renaynay: user input?
```

Comment:
What does the comment refer to?

Reply:
I'm passing nil here for `userInput` parameter, but not really sure about it.

Reply:
I think this is to get the user's password for the file: https://github.com/cosmos/cosmos-sdk/blob/20735e792c85a0d84ca32e97a0a3c814bcc3529f/crypto/keyring/keyring.go#L671

Reply:
Can we manually test this before merging? I think in the SDK this is used with Stdin. The userInput is fetched from this context:
https://github.com/cosmos/cosmos-sdk/blob/fece3f0485dd418a2bc54ecc61f17da3303192c7/simapp/simd/cmd/root.go#L46

Reply:
I manually tested it and it does not panic.

Reply:
what is the userInput used for?

Reply:
To get the user's input for the password, which is used to encrypt the file. If it does not panic, it's OK for now. We should probably support all the other backends as well though.